csrc/cpu/cpu_attn_macros.h

```diff
@@ -0,0 +1,63 @@
+#ifndef CPU_ATTN_MACROS_H
+#define CPU_ATTN_MACROS_H
+
+// x86_64
+#ifdef __x86_64__
+  #define FAST_SPINNING _mm_pause();
+
+  #ifdef __AVX512F__
+    #define DEFINE_FAST_EXP                                                    \
```

Comment:
what's stopping us from having this in `cpu_types_x86.hpp`? the same implementation in pytorch lives in the vectorization abstractions under `pytorch/aten/src/ATen/cpu/vec/vec512/vec512_float.h`